core/chaincode/handler_test.go

```diff
@@ -2559,7 +2559,7 @@ var _ = Describe("Handler", func() {
 			Eventually(fakeChatStream.SendCallCount).Should(Equal(1))
 			Consistently(fakeChatStream.SendCallCount).Should(Equal(1))
 			msg := fakeChatStream.SendArgsForCall(0)
-			Expect(msg).To(Equal(&expectedMessage))
+			Expect(msg).To(Equal(expectedMessage))
```

Comment:
The standard Gomega `Equal` matcher is not safe for use on protobuf messages at all. It simply does a deep equal comparison, and this is not valid for protobuf messages since they can be logically equal while also having some implementation fields with different values.

Some other parts of the codebase use `proto.Equal` for protobuf message comparison, and then use Gomega to _Expect_ a True return value. I think a better approach is to implement a matcher specifically for protobuf messages, for example:

```go
func ProtoEqual(expected proto.Message) types.GomegaMatcher {
	return &protoEqualMatcher{
		expected: expected,
	}
}

type protoEqualMatcher struct {
	expected proto.Message
}

func (m *protoEqualMatcher) Match(actual interface{}) (bool, error) {
	actualMessage, ok := actual.(proto.Message)
	if !ok {
		return false, fmt.Errorf("ProtoEqual expects a proto.Message, got a %T", actual)
	}

	return proto.Equal(m.expected, actualMessage), nil
}

func (m *protoEqualMatcher) FailureMessage(actual interface{}) string {
	if message, ok := actual.(proto.Message); ok {
		return fmt.Sprintf("Expected\n\t%s\nto equal\n\t%s", message, m.expected)
	}

	return fmt.Sprintf("Expected\n\t%#v\nto equal\n\t%s", actual, m.expected)
}

func (m *protoEqualMatcher) NegatedFailureMessage(actual interface{}) string {
	if message, ok := actual.(proto.Message); ok {
		return fmt.Sprintf("Expected\n\t%s\nnot to equal\n\t%s", message, m.expected)
	}

	return fmt.Sprintf("Expected\n\t%#v\nnot to equal\n\t%s", actual, m.expected)
}
```

This is then used as follows:
```go
Expect(actualMsg).To(ProtoEqual(expectedMsg))
```

Reply:
fix